### PR TITLE
add dim checks

### LINKED ISSUE #19: Check input dimensions
@jbusecke  pointed out that `time` is assumed to be the first dimension of the input DataArray. 

So we need the following input checks :
- [ ]  transpose dimensions so that they are ('time',  'y', 'x') 
- [ ] raise a value error if there are more/less than 3 dimensions. 

Here is a possible solution:

```python 
timedim = 'time'
xdim = 'lon'
ydim = 'lat'

if ((timedim, ydim, xdim) != da.dims):
    try:
        da = da.transpose(timedim, ydim, xdim) 
    except:
        raise ValueError('Ocetrac currently only supports 3D DataArrays. The dimensions should only contain time, x, and y positions.')

```

### PR DESCRIPTION
**Pull request by hscannell**
Added a `timedim` parameter and additional checks for make sure that the dimensions are (time, y, x).

### CODE REVIEW

File: ocetrac/model.py [hunk 1]
```diff
@@ -44,8 +47,16 @@ def __init__(self, da, mask, radius, min_size_quartile, xdim, ydim):
         self.mask = mask
         self.radius = radius
         self.min_size_quartile = min_size_quartile
+        self.timedim = timedim
         self.xdim = xdim
         self.ydim = ydim   
+        
+        if ((timedim, ydim, xdim) != da.dims):
```
Comment (jbusecke) [inline ocetrac/model.py:54]: Dont you have to wrap this in `set(...)` to order it?

Comment (jbusecke) [reply ocetrac/model.py:54]: Nvm, i misunderstood the code.

File: ocetrac/model.py [hunk 2]
```diff
@@ -44,8 +47,16 @@ def __init__(self, da, mask, radius, min_size_quartile, xdim, ydim):
...
+            try:
+                da = da.transpose(timedim, ydim, xdim) 
+            except:
+                raise ValueError(f'Ocetrac currently only supports 3D DataArrays. The dimensions should only contain ({timedim}, {xdim}, and {ydim}). Found {list(ds.dims)}')
```
Comment (jbusecke) [inline ocetrac/model.py]: Ugh sorry I committed directly, so feel free to revert. This basically just uses the actual names given for `timedim` etc.

File: ocetrac/model.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def __init__(self, da, mask, radius, min_size_quartile, timedim, xdim, ydim):
             try:
                 da = da.transpose(timedim, ydim, xdim) 
             except:
-                raise ValueError('Ocetrac currently only supports 3D DataArrays. The dimensions should contain time, x, and y.')
+                raise ValueError(f'Ocetrac currently only supports 3D DataArrays. The dimensions should only contain ({timedim}, {xdim}, and {ydim}). Found {list(ds.dims)}')
```
Comment (hscannell) [inline ocetrac/model.py]: Nice!

**Review by jbusecke — CHANGES_REQUESTED** [review]
This looks very good. I would however add a test to check that the value error is raised. You can just pass some simple datarray that has dimensions of `x`, `y`, `time`, `additional` and check that the error is properly raised.

File: ocetrac/model.py (hunk 1 above)
Comment (hscannell) [reply ocetrac/model.py:54]: Done!
```python 
assert ((timedim, ydim, xdim) != Anom.expand_dims('level').dims)
````

**Pull request merged**